### PR TITLE
Material ui study

### PR DESCRIPTION
**Pull request by jackbsteinberg**

### CODE REVIEW

File: study-group/Toast-Implementation-Studies.md [hunk 1]
```diff
@@ -79,8 +79,121 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
     - How to properly handle top layer is an ongoing debate (see [here](https://github.com/whatwg/html/issues/897#issuecomment-198512778))
 
 ## [Material-UI Snackbar](https://material-ui.com/api/snackbar/)
+Material-UI Snackbar is a component from Google's Material-UI React library that provides an interactive toast notification.
+
+### Design & Sample Code
+"Snackbar" is the name used in many Google component libraries for an extensible, interactable toast component.
+Unlike Google *Toasts*, the Snackbar can have a call to action and a close button, alongside other differentiating properties.
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]: semantic breaks for this file too. If you're wondering how aggressively to break things, take a look at

https://github.com/w3c/csswg-drafts/blob/master/css-shadow-parts-1/Overview.bs

File: study-group/Toast-Implementation-Studies.md [hunk 2]
```diff
@@ -79,8 +79,121 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
...
+/>
+```
+
+### Notable Features & Details
+
+#### SnackbarContent
+Material-UI provides both `<Snackbar>` and `<SnackbarContent>` components.
+The SnackbarContent lets the developer style the `root`, `message`, and `action` via the `classes` property,
+and the component is used as a child of Snackbar to display when style is needed.
+The `classes` property of Snackbar lets the developer add style the `root`,
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]:
```suggestion
The `classes` property of Snackbar lets the developer style the `root`,
```

???

File: study-group/Toast-Implementation-Studies.md [hunk 3]
```diff
@@ -79,8 +79,121 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
...
+and each of the possible positions the toast can be in (`anchorOriginTopCenter`, `anchorOriginBottomCenter`, etc.).
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md:183]: this clause doesn't seem to connect to the previous one.

File: study-group/Toast-Implementation-Studies.md [hunk 4]
```diff
@@ -79,8 +79,121 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
...
+This is likely to allow finer control over the exact placement in each position.
+
+#### resumeHideDuration
+The Snackbar component has property that allows you to reset the timeout to a specified number of milliseconds after user interaction.
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]:
```suggestion
The Snackbar component has a property that allows you to reset the timeout to a specified number of milliseconds after user interaction.
```

???

File: study-group/Toast-Implementation-Studies.md [hunk 5]
```diff
@@ -407,10 +520,105 @@ The library allows showing multiple toasts, and gives the developer specific con
 - Comes with default icons, can change by modifying CSS.
 - **Note**: On mobile there's currently no simple way to stop the timeout, accomplished by hover on desktop.
 
-## Bootstrap
+## [Bootstrap Toast](https://getbootstrap.com/docs/4.2/components/toasts/)
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]: Somehow the bootstrap study is in here too. Rebase? Or is it me not knowing how to use github?

Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: Hmm you're right, I'm not really sure how I got that wrong. How would you recommend I solve this (my Git-Fu is somewhat lacking at the moment)?

File: study-group/Toast-Implementation-Studies.md (hunk 1 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: On it

File: study-group/Toast-Implementation-Studies.md (hunk 5 above)
Comment (domenic) [reply study-group/Toast-Implementation-Studies.md]: I'm awake now, and can help in person :)

Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: Got it rebased, thanks!

File: study-group/Toast-Implementation-Studies.md [hunk 6]
```diff
@@ -74,21 +74,137 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
...
 
 ## [Material-UI Snackbar](https://material-ui.com/api/snackbar/)
+Material-UI Snackbar is a component from Google's Material-UI React library that provides an interactive toast notification.
+
+### Design & Sample Code
+"Snackbar" is the name used in many Google component libraries for an extensible, interactable toast component.
+Unlike Google *Toasts*, the Snackbar can have a call to action and a close button,
+alongside other differentiating properties.
+
+A Material-UI Snackbar is defined like so:
```
Comment (domenic) [inline study-group/Toast-Implementation-Studies.md]:
```suggestion
A Material-UI Snackbar is used like so:
```

File: study-group/Toast-Implementation-Studies.md [hunk 7]
```diff
@@ -74,21 +74,137 @@ The `restoreFocusOnClear` attribute restores page's focus once the toast is clea
...
+The `classes` property of Snackbar lets the developer style the `root`,
+and each of the possible positions the toast can be in (`anchorOriginTopCenter`, `anchorOriginBottomCenter`, etc.).
+This is likely to allow finer control over the exact placement in each position.
+
+#### resumeHideDuration
+The Snackbar component has a property that allows the developer to reset the timeout to a specified number of milliseconds after user interaction.
+
+#### notistack
+It is not natively supported to display multiple Snackbars at a time, 
+but the spec recommends using the [notistack](https://github.com/iamhosseindhv/notistack) library to make it possible.
```
Comment (domenic) [inline study-group/Toast-Implementation-Studies.md]:
```suggestion
but the documentatoin recommends using the [notistack](https://github.com/iamhosseindhv/notistack) library to make it possible.
```

I think?

**Review by domenic — APPROVED** [review]
LGTM with nits

File: study-group/Toast-Implementation-Studies.md (hunk 7 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: You're right, I wasn't sure what the difference was at the time
```suggestion
but the documentation recommends using the [notistack](https://github.com/iamhosseindhv/notistack) library to make it possible.
```

**Pull request merged**